placement/algo/mirrored.go

```diff
@@ -285,6 +267,83 @@ func allInstancesInState(
 
 	return len(instanceIDs) == 0
 }
+
+func (a mirroredAlgorithm) returnInitializingShards(
+	p placement.Placement,
+	instanceIDs []string,
+) (placement.Placement, error) {
+	shouldContinue := true
+	for shouldContinue {
```

Comment:
nit: might be easier to understand to write it as (also this narrows the scope of `madeProgress`)
```
for {
  madeProgress := false
  for ... {
   }
   if !madeProgress {
      break
   }
}
```